src/codewhisperer/service/classifierTrigger.ts

```diff
@@ -44,7 +43,7 @@ export class ClassifierTrigger {
         Enter: 0.2853,
     }
 
-    private languageCoefficientMap: Readonly<Record<string, number>> = {
+    private languageCoefficientMap: Readonly<Record<CodewhispererLanguage, number>> = {
```

Comment:
so whenever we add a new language we have to also update the language coefficient here? That's not super ideal because new languages usually won't be supported by classifier trigger yet

Reply:
but the good side is the compiler will know in advance if you have typo

Reply:
which will not be captured if we use plain string

Reply:
i don't think it will add much burden when we add a new language tho

Reply:
can we make it `Partial<CodewhispererLanguage>`?

Reply:
yes that sgtm, it's even better in terms of keeping things easy and prevent error

Reply:
it's not burden it's that adding coefficient needs to be deliberate

Reply:
yes, but we should use an unified interface for programming language instead plain string, which makes things error prone

Reply:
Zoe I reverted this change for now to make this PR only change 1 aspect FileContext for easier review, let's follow up on the remaining part step by step